sql-to-dbsp-compiler/lib/sqllib/src/variant.rs

```diff
@@ -120,69 +120,120 @@ impl<D: Fallible> rkyv::Deserialize<Box<dyn StructVariant>, D> for Box<dyn Struc
 pub enum Variant {
     #[default]
     SqlNull,
-    Boolean(Option<bool>),
-    TinyInt(Option<i8>),
-    SmallInt(Option<i16>),
-    Int(Option<i32>),
-    BigInt(Option<i64>),
-    Real(Option<F32>),
-    Double(Option<F64>),
-    Decimal(Option<Decimal>),
-    String(Option<String>),
-    Date(Option<Date>),
-    Time(Option<Time>),
-    Timestamp(Option<Timestamp>),
-    ShortInterval(Option<ShortInterval>),
-    LongInterval(Option<LongInterval>),
-    Geometry(Option<GeoPoint>),
+    VariantNull,
+    Boolean(bool),
+    TinyInt(i8),
+    SmallInt(i16),
+    Int(i32),
+    BigInt(i64),
+    Real(F32),
+    Double(F64),
+    Decimal(Decimal),
+    String(String),
+    Date(Date),
+    Time(Time),
+    Timestamp(Timestamp),
+    ShortInterval(ShortInterval),
+    LongInterval(LongInterval),
+    Geometry(GeoPoint),
     #[size_of(skip, skip_bounds)]
-    Array(#[omit_bounds] Option<Vec<Variant>>),
+    Array(#[omit_bounds] Vec<Variant>),
     #[size_of(skip, skip_bounds)]
-    Map(#[omit_bounds] Option<BTreeMap<String, Variant>>),
+    Map(#[omit_bounds] BTreeMap<Variant, Variant>),
+    // Special case for string maps
+    // StringMap(#[omit_bounds] BTreeMap<String, Variant>),
     //Struct(Option<Box<dyn StructVariant>>),
 }
 
-// A macro for TryInto<Option<T>> for Variant
-macro_rules! try_into_option {
-    ($variant:ident, $type:ty) => {
-        impl TryInto<Option<$type>> for Variant {
-            type Error = &'static str;
+impl Variant {
+    fn getTypeString(&self) -> String {
```

Comment:
clippy doesn't complain about this?
```suggestion
    fn type_string(&self) -> String {
```

this should probably return &'static str